### PR TITLE
HHT-669: knowledge graph exporting based in k8s resources and metrics

### PR DESCRIPTION
**Pull request by ktatarnikovhiro**

### CODE REVIEW

**Review by vladimir-gladkii — APPROVED** [review]
Code looks nice

File: app/clients/k8s/test_k8s_client.py
Comment (a-zharinov) [inline app/clients/k8s/test_k8s_client.py:1]: Konstantin, we use pytest, so I recommend to use a pytest style.

```
import asyncio
from unittest import TestCase

from app.clients.k8s.k8s_client import ResourceSnapshot
from app.clients.k8s.mock_k8s_client import MockK8SClient

```
import asyncio

import pytest

from app.clients.k8s.k8s_client import ResourceSnapshot
from app.clients.k8s.mock_k8s_client import MockK8SClient


@pytest.fixture
def client() -> MockK8SClient:
    client = MockK8SClient()
    client.mock_daemonsets([{"daemonsets": "fake"}])
    client.mock_deployments([{"deployments": "fake"}])
    client.mock_jobs([{"jobs": "fake"}])
    client.mock_nodes([{"nodes": "fake"}])
    client.mock_pods([{"pods": "fake"}])
    client.mock_replicasets([{"replicasets": "fake"}])
    client.mock_statefullsets([{"statefullsets": "fake"}])
    return client


def test_get_resource_snapshot(client) -> None:
    actual = asyncio.run(client.fetch_snapshot())
    expected = ResourceSnapshot(
        pods=[{"pods": "fake"}],
        nodes=[{"nodes": "fake"}],
        deployments=[{"deployments": "fake"}],
        jobs=[{"jobs": "fake"}],
        statefullsets=[{"statefullsets": "fake"}],
        daemonsets=[{"daemonsets": "fake"}],
        replicasets=[{"replicasets": "fake"}],
    )
    assert expected == actual
```

File: app/kgexporter_context.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        self.updater = KGUpdater(self.terminated, self.queue, kg_repository)
+        self.runner = asyncio.Runner()
+        self.tasks = []
+
+    def start(self) -> None:
+        if self.terminated.is_set():
+            return
+        self.terminated.clear()
+        self.runner.run(self.run_tasks())
+        server, _ = start_http_server(self.settings.prometheus.endpoint_port)
```
Comment (a-zharinov) [inline app/kgexporter_context.py:63]: As we discussed, please check it you can runtime serves via asyncio (not in a new thread).

**Review by a-zharinov — APPROVED** [review]

File: app/util/clock.py
[the remark this answers is not in the document]
Comment (ktatarnikovhiro) [reply app/util/clock.py:1]: yes, thanks, I am aware about this approach but somehow overlooked it in between all functional requirements. We can fix this as one of the next cleanups.

File: app/clients/k8s/test_k8s_client.py
Comment (ktatarnikovhiro) [reply app/clients/k8s/test_k8s_client.py:1]: pytest is used where necessary.

File: app/kgexporter_context.py (hunk 1 above)
Comment (ktatarnikovhiro) [reply app/kgexporter_context.py]: fixed

**Pull request merged**